Modules/Loadable/TortuosityModule/Logic/vtkSlicerTortuosityLogic.cxx

```diff
@@ -262,13 +266,21 @@ ::RunMetrics(vtkMRMLSpatialObjectsNode* node, int flag)
       continue;
       }
 
+    VesselTubeType::Pointer smoothedTube = VesselTubeType::New();
+    VesselTubeType::Pointer subsampledTube = VesselTubeType::New();
+
+    // Smooth the vessel
+    smoothedTube = tube::SmoothTube<VesselTubeType>(currTube, smoothingScale, (tube::SmoothTubeFunctionEnum)smoothingMethod );
+    // Subsample the vessel
+    subsampledTube = tube::SubsampleTube<VesselTubeType>(smoothedTube, subsampling);
+
     // Update filter
     FilterType::Pointer filter = FilterType::New();
     filter->SetMeasureFlag(flag);
-    filter->SetInput(currTube);
+    filter->SetInput(subsampledTube);
     filter->Update();
 
-    if (filter->GetOutput()->GetId() != currTube->GetId())
+    if (filter->GetOutput()->GetId() != subsampledTube->GetId())
```

Comment:
Good point.

Go with that for vtk-derived classes.

On Wed, Apr 8, 2015 at 11:13 AM, Jean-Christophe Fillion-Robin <
notifications@github.com> wrote:

> In Modules/Loadable/TortuosityModule/Logic/vtkSlicerTortuosityLogic.cxx
> <https://github.com/KitwareMedical/VesselView/pull/96#discussion_r27979205>
> :
>
> >      filter->Update();
> >
> > -    if (filter->GetOutput()->GetId() != currTube->GetId())
> > +    if (filter->GetOutput()->GetId() != subsampledTube->GetId())
>
> Should VTK derived class following VTK style ?
> See
> https://docs.google.com/a/kitware.com/document/d/1nzinw-dR5JQRNi_gb8qwLL5PnkGMK2FETlQGLr10tZw/edit
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/KitwareMedical/VesselView/pull/96/files#r27979205>.
>



-- 
====================================================
Stephen R. Aylward, Ph.D.
Senior Director of Operations, North Carolina, Kitware, Inc.
http://www.kitware.com and http://www.aylward.org
(919) 969-6990 x300
====================================================